packages/vike-react-apollo/README.md

```diff
@@ -0,0 +1,130 @@
+<!-- WARNING: keep links absolute in this file so they work on NPM too -->
+
+[<img src="https://vike.dev/vike-readme.svg" align="right" height="90">](https://vike.dev)
+[![npm version](https://img.shields.io/npm/v/vike-react-apollo)](https://www.npmjs.com/package/vike-react-apollo)
+
+# `vike-react-apollo`
+
+[Apollo GraphQL](https://www.apollographql.com/docs/react/) integration for [vike-react](https://github.com/vikejs/vike-react/tree/main/packages/vike-react).
+
+`vike-react-apollo` enables you to create components that can fetch data.
+
+You can use it instead of [Vike's `data()` hook](https://vike.dev/data): with `vike-react-apollo` you fetch data at component-level instead of page-level.
+
+You also get:
+ - [Progressive rendering](https://vike.dev/streaming#progressive-rendering) for a significant (perceived) increase in page speed.
+ - Fallback component upon loading/error. (See [`withFallback()`](#withfallback).)
+ - Caching. ([Read more](https://www.apollographql.com/docs/react/caching/cache-configuration).)
+
+See [example](https://github.com/vikejs/vike-react/tree/main/examples/apollo).
+
+> [!NOTE]  
+> `vike-react-apollo` leverages [React 18's suspense streaming feature](https://github.com/brillout/react-streaming#readme). (Similar to [Next.js Loading UI and Streaming](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming), but on a component level.)
+
+
+## Installation
+
+1. `pnpm i @apollo/client @apollo/client-react-streaming graphql vike-react-apollo`
+2. Extend `+config.ts`:
+   ```ts
+   // /pages/+config.ts
+
+   import type { Config } from 'vike/types'
+   import vikeReact from 'vike-react/config'
+   import vikeReactApollo from 'vike-react-apollo/config'
+
+   export default {
+     ...
+     extends: [vikeReact, vikeReactApollo]
+   } satisfies Config
+   ```
+3. Create `+ApolloConfig.ts`:
```

Comment:
This currently only defines the Apollo client. If `new ApolloClient()` doesn't accept any other argument then maybe we can rename the setting to `+ApolloClient`.

In order to keep things as zero-config as possbile, maybe be can also support this for those who don't need `pageContext`:

```ts
// pages/+ApolloConfig.ts

import { InMemoryCache } from '@apollo/client-react-streaming'

export default {
  uri: 'https://countries.trevorblades.com',
  cache: new InMemoryCache()
}
```

Do we actually foresee a use case for using `pageContext` here? If we don't foresee one, then I'd say let's remove it and add it only after a user complains.

Reply:
It needs to be a function, because the config file is only loaded once, but we need a new cache for each request server-side. PageContext is useful for proxying headers/cookies for authentication. For example the user can add an `ApolloLink` that reads the headers/cookies from the `pageContext` and sets it on the graphql request server-side.

Reply:
I see, makes sense.

How about this:

```js
import { InMemoryCache, ApolloClient } from '@apollo/client-react-streaming'

export default (pageContext) =>
  new ApolloClient({
    uri: 'https://countries.trevorblades.com',
    cache: new InMemoryCache()
  })
```